src/arrayMethodJoin.js

```diff
@@ -4,8 +4,21 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let result = '';
+
+    for (let i = 0; i < this.length; i++) {
+      if (this[i] !== undefined && this[i] !== null) {
+        result += `${this[i]}`;
+      }
+
+      if (i !== this.length - 1) {
+        result += `${separator}`;
```

Comment:
The `continue` statement is unnecessary, as it will automatically move to the next iteration without it.